### ISSUE TITLE
transform on elements is not independent of parent group's transform

### ISSUE THREAD

**Issue by lweigand** [wout/svg.js#359]
What should happen:
The following code should draw two polygons, one of which is mirrored. Through the transform on the group, both of them should be offset  by 100/100

````javascript
var draw = SVG('wrapper').size(600, 600);
var group = draw.group().move(100, 100)
var poly1 = group.polygon('0,0 100,-50 50,50 0,100')
var poly2 = group.polygon('0,0 100,-50 50,50 0,100')
poly2.scale(-1)
```

What actually happens:
Somehow, `poly2` takes the transform of `group` into account and is offset by setting scale to -1.

The problem does not occur when either the transform on the group is applied after everything else or there is no transform at all.

Maybe I am just overseeing something in which case I'd be sorry for raising a new issue.

**Comment by Fuzzyma** [comment]
For some reason transform gets the current transformation-matric not from its transform-attr but from the method `getCTM()` which returns a Matrix to transform the Elements coord system back to the root-svg system.

To give you a quickfix you can do one of these:

- as you mentioned apply the transform in the right order
- move the poly back after transforming
- change #L2017 to `new SVG.Matrix(target.attr('transform'))`

The last one would be my fix for this in general but I have to test that properly (even if no specs fail for now)

**Comment by lweigand** [comment]
You are right, the transform is calculated in context of the root coordinate system.
The change suggested by you does indeed fix the issue, would be happy to see this merged into master.

To be honest, I was a bit suprised to come across such a strange issue as svg.js has been around for quite some time and and seems to be quite mature.
No criticism, I really appreciate the work of everyone who has contributed to this project, it is very well-thought-out and I like the overall design very much.

**Comment by Fuzzyma** [comment]
Version 2 got a completely reworked transform-system. It was released a few weeks ago. That explains such bugs. And since transformation is a very complicated issue I dont like to change something there.
However that has to be fixed so I will try my best :D

**Comment by Fuzzyma** [comment]
This should be fixed in 2.0.5 now.
The transform-method now uses the transformations specified in the trasnform attribute instead of the ctm()
